examples/geometric_art.rs

```diff
@@ -0,0 +1,127 @@
+use turtle::{random, random_range, Color, Turtle};
+
+fn main() {
+    // Parameters to play around with for changing the character of the created drawing.
+    let width = 800;
+    let height = 600;
+    let row_count = 3;
+    let col_count = 4;
+    let color_count = 7;
+
+    let mut turtle = ArtisticTurtle::new(width, height, row_count, col_count, color_count);
+    turtle.draw();
+}
+
+/// A special kind of turtle that is specialized in creating drawings of colorful triangles.
+struct ArtisticTurtle {
+    turtle: Turtle,
+    row_count: u32,
+    col_count: u32,
+    row_height: f64,
+    col_width: f64,
+    colors: Vec<Color>,
+}
+
+impl ArtisticTurtle {
+    /// Create a new turtle using the given parameters.
+    /// `width`: total width of the drawing
+    /// `height`: total height of the drawing
+    /// `row_count`: number of rows of triangles to be created
+    /// `col_count`: number of columns per row to be created
+    /// `color_count`: number of different colors to use
+    fn new(width: u32, height: u32, row_count: u32, col_count: u32, color_count: u32) -> Self {
+        // Create and prepare the turtle.
+        let mut turtle = Turtle::new();
+        turtle.set_speed("instant");
+        turtle.pen_up();
+        turtle.drawing_mut().set_size([width, height]);
+
+        // Move turtle to the upper left corner of the drawing.
+        let size = turtle.drawing().size();
+        let x_start = size.width / 2;
+        let y_start = size.height / 2;
+        turtle.go_to([f64::from(x_start) * -1.0, f64::from(y_start)]);
+        turtle.set_speed(20);
+
+        // Calculate height and width of the triangles.
+        let row_height = f64::from(size.height) / f64::from(row_count);
+        let col_width = f64::from(size.width) / f64::from(col_count);
+
+        // Prepare a set of random colors to randomly choose from when drawing.
+        let mut colors = Vec::new();
```

Comment:
You definitely don't have to change this code to use this (it's probably too advanced/over complicated), but in case you're curious, you can actually replace this whole loop with just:

```rust
let colors: Vec<Color> = from_fn(|| Some(random())).take(color_count).collect();
```

This uses [`from_fn`](https://doc.rust-lang.org/std/iter/fn.from_fn.html) which was added in Rust 1.34. Here's a working example of this line in [the Rust playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=06e3c193b735762e3aa11046424a10c5).

Reply:
No, this is not too advanced at all! I think being able to create such iterator chains is one of the coolest features of Rust. :) I'm just not that used to making them up in my mind yet, so I tend to fall back to using the good old for loop I know from other languages. Thanks for pointing this out!